collect/src/main/kotlin/com/certora/collect/HashTreapSet.kt

```diff
@@ -228,8 +229,10 @@ internal class HashTreapSet<@Treapable E>(
         }
     }.iterator()
 
-    override fun shallowGetSingleElement(): E? = element.takeIf { next == null }
-
+    override fun singleOrNull(): E? = element.takeIf { next == null && left == null && right == null }
+    override fun single(): E = element.also {
+        if (next != null || left != null || right != null) { throw IllegalArgumentException("Set contains more than one element") }
+    }
```

Comment:
just because you *can* write something as a one liner, doesn't mean you should.

This "post fixing" conditional stuff *sucks* in ruby, let's not bring it into kotlin :)

Reply:
Fine. :)